src/main/java/com/linkedin/kmf/KafkaMonitor.java

```diff
@@ -98,19 +114,26 @@ private void checkHealth() {
     }
   }
 
-  public void stop() {
+  public synchronized void stop() {
```

Comment:
Ah I get your previous question wrong. You are asking about start() not stop().

I would recommend to make the implementation reasonable and simple until there is use-case to justify a more complex semantics. Therefore I think start() means, if not running, then initialize state and set _isrunning to true; and stop() means, if runnning, then tear down everything and set _isrunning to false. If user calls stop() before start(), then stop() is no-op and start() will start things. User needs to call stop() after start() to stop the Kafka Monitor. Does this make sense?